### PR TITLE
Refactor `toolchain` package

### PR DESCRIPTION
**Pull request by mchernyavsky**

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/toolchain/components/Cargo.kt [hunk 1]
```diff
@@ -48,6 +52,14 @@ import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.Paths
 
+fun RsToolchain.rawCargo(): Cargo = Cargo(this)
```
Comment (Undin) [inline src/main/kotlin/org/rust/cargo/toolchain/components/Cargo.kt:55]: Do we really need `raw` prefix here? I have never understood why it's needed. It looks rather inconsistent with other tools and doesn't add additional info (at least for me). Maybe drop it while this refactoring as well?

**Review by Undin — COMMENTED** [review]
Looks ok in general
But I'm not sure we need separate packages for binaries and components. What's the main difference between them? Maybe it's worth merging `binaries` and `components` into a single package?

**Comment by mchernyavsky** [comment]
bors try

**Comment by mchernyavsky** [comment]
@Undin Done.

**Review by mchernyavsky — COMMENTED** [review]
bors r=Undin

**Comment by Undin** [comment]
bors retry

**Pull request merged**